### PR TITLE
PIPE-91 Add Login page

### PR DESCRIPTION
**Pull request by tomerdayan168**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -107,4 +107,6 @@ dist
 
 
 # DB Files
-*.db
\ No newline at end of file
+*.db
+
+package-lock.json
```
Comment (San-Moshe) [inline .gitignore]: I think it should not be ignored.
https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

File: components/login.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+import LoginView from "./login-view";
+import { apiPost } from "../utils/http-calls";
+import { useRouter } from "next/router";
+import { HOME_ROUTE } from "../config";
+
+export default function Login() {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [failed, setFailed] = useState(true);
```
Comment (San-Moshe) [inline components/login.js]: Avoid negative names for booleans to increase readability.

File: components/login.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      await apiPost("login", {
+        username: username,
+        password: password,
+      });
+      router.push(HOME_ROUTE);
+    } catch (error) {
+        setFailed(false)
+        console.log(error);
+    }
+  };
```
Comment (San-Moshe) [inline components/login.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#handling_rejected_promises

File: pages/_middleware.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  const newToken = jwt.sign({ username: payload.username }, JWT_KEY, {
+    algorithm: "HS256",
+    expiresIn: JWT_EXPIRATION,
+  });
+
+  let res = NextResponse.next()
+  res.cookie("token", newToken, { maxAge: JWT_EXPIRATION * 1000 });
+
+  return res;
+}
```
Comment (San-Moshe) [inline pages/_middleware.js]: https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/

File: pages/api/login.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  const username = reqBody["username"];
+  const password = reqBody["password"];
+
+  if (validate(username, password)) {
+    const token = jwt.sign({ username }, JWT_KEY, {
+      algorithm: "HS256",
+      expiresIn: JWT_EXPIRATION,
+    });
+
+    res.setHeader("Set-Cookie", serialize("token", token, { path: "/" }));
```
Comment (San-Moshe) [inline pages/api/login.js:29]: Why sending token via response header?

Comment (San-Moshe) [reply pages/api/login.js:29]: I'm not saying this is wrong

**Review by San-Moshe — CHANGES_REQUESTED** [review]

File: pages/api/login.js (hunk 5 above)
Comment (tomerdayan168) [reply pages/api/login.js:29]: Because the res object doesn't have the option to set the token in the cookie (function doesn't exist)

**Review by San-Moshe — APPROVED** [review]

**Pull request merged**